src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -0,0 +1,408 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.security;
+
+import jdk.internal.javac.PreviewFeature;
+
+import sun.security.pkcs.PKCS8Key;
+import sun.security.rsa.RSAPrivateCrtKeyImpl;
+import sun.security.util.Pem;
+
+import javax.crypto.EncryptedPrivateKeyInfo;
+import javax.crypto.spec.PBEKeySpec;
+import java.io.*;
+import java.security.cert.*;
+import java.security.spec.*;
+import java.util.Base64;
+import java.util.Objects;
+
+/**
+ * {@code PEMDecoder} is an immutable class for Privacy-Enhanced Mail (PEM)
+ * data.  PEM is a textual encoding used to store and transfer security
+ * objects, such as asymmetric keys, certificates, and certificate revocation
+ * lists (CRL).  It is defined in RFC 1421 and RFC 7468.  PEM consists of a
+ * Base64-formatted binary encoding enclosed by a type-identifying header
+ * and footer.
+ *
+ * <p> Decoding methods return an instance of a class that matches the data
+ * type and implements {@link DEREncodable} unless specified. The following
+ * types are decoded into Java Cryptographic Extensions (JCE) object:
+ * <pre>
+ *     PRIVATE KEY, RSA PRIVATE KEY, PUBLIC KEY, CERTIFICATE,
+ *     X509 CERTIFICATE, X509 CRL, and ENCRYPTED PRIVATE KEY.
+ * </pre>
+ *
+ * A specified return class must implement {@link DEREncodable} and be an
+ * appropriate JCE object class for the PEM; otherwise an
+ * {@link IllegalArgumentException} is thrown.
+ *
+ * <p> If the PEM does not have a corresponding JCE object, it returns a
+ * {@link PEMRecord}. Any PEM can be decoded into a {@code PEMRecord} if the
+ * class is specified. Decoding a {@code PEMRecord} returns corresponding
+ * JCE object or throws a {@link IllegalArgumentException} if no object is
+ * available.
+ *
+ * <p> A new immutable {@code PEMDecoder} instance is created by using
+ * {@linkplain #withFactory} and/or {@linkplain #withDecryption}.  Configuring
+ * an instance for decryption does not prevent decoding with unencrypted PEM.
+ * Any encrypted PEM that does not use the configured password will throw an
+ * {@link SecurityException}. A decoder instance not configured with decryption
+ * returns an {@link EncryptedPrivateKeyInfo} with encrypted PEM.
+ * EncryptedPrivateKeyInfo methods must be used to retrieve the
+ * {@link PrivateKey}.
+ *
+ * <p> {@code String} values returned by this class use character set
+ * {@link java.nio.charset.StandardCharsets#ISO_8859_1 ISO-8859-1}
```

Comment:
`String` values in Java are always encoded in UTF-16. Did you mean to write something like, "Byte streams consumed by methods in this class are assumed to represent characters encoded in the ISO-8859-1 charset" ?